### PR TITLE
Fix None gradient to enable maximum-likelihood training. This fixes issue #59.

### PR DESCRIPTION
**Pull request by yiboyang**
Fix None gradient to enable maximum-likelihood training.

This PR fixes resolves issue #59. Previously, likelihood-based training fails because the gradient of the log probability is None even when calling `ODESolver.compute_likelihood` with `enable_grad=True`. This PR fixes the issue by leaving out `.detach()` statements when `enable_grad=True`. The default behavior (with `enable_grad=False`) is unaffected.

### CODE REVIEW

**Comment by timonpalm** [comment]
I can confirm that this fix worked. In the plot, you can see the likelihoods of a grid of points computed by the flow (A), their gradients using the current state of the repo (B) and the gradients with this fix.

image

It is apparent that (B) is clearly wrong. 

Wish this PR to be merged soon. :)

**Comment by yiboyang** [comment]
The formatting issue should be fixed now with 04e51659db4c59b4f7a9e58b80c925c0a16c7c20.
